hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1203 @@
+# See LICENSE file for full copyright and licensing details.
+
+import time
+from datetime import datetime, timedelta
+
+import pytz
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as dt
+
+
+class HotelFolio(models.Model):
+
+    _inherit = "hotel.folio"
+    _order = "reservation_id desc"
+
+    reservation_id = fields.Many2one(
+        "hotel.reservation", "Reservation", ondelete="restrict"
+    )
+
+    def write(self, vals):
+        context = dict(self._context)
+        context.update({"from_reservation": True})
+        res = super(HotelFolio, self).write(vals)
+        reservation_line_obj = self.env["hotel.room.reservation.line"]
+        for folio in self:
+            reservations = reservation_line_obj.search(
```

Comment:
To be sure, We should have relation b/w `hotel.room.reservation.line` and `reservation_id` (Add it so we can remove this bad hack)

Reply:
Absolutely right.As of now, we will keep as same but I will discuss with team and surely we will improve such kinds of code.